Bug/safari crash on signout

leakymemory opened the pull request:
Fixes #129 

On clipper sign-out, we were listening for the response URL and then manually closing the popup window.  However, it looks like on sign-out, this isn't necessary as the window gets closed automatically.

This causes a problem because the listener we set up gets called and tries to access the newWindow object, which is now null.  In pre-Sierra versions of Safari this seemed to be ok, but in Sierra it now causes the browser to crash.

leakymemory committed:
Make the popup listener an option on the Safari launch popup method.

leakymemory committed:
Cleanup

mttwc commented on src/scripts/extensions/safari/safariWorker.ts:
```diff
@@ -129,34 +129,37 @@ export class SafariWorker extends ExtensionWorkerBase<SafariBrowserTab, SafariBr
-			};
-			newWindow.addEventListener("close", closeListener);
+						newWindow.removeEventListener(redirectListener);
+						newWindow.close();
```
Shouldn't we wrap the new if statement around the .close() call? That way we still get our error reporting.

leakymemory replied:
Yeah, I thought about that, though in the waitForClose option implies that we cannot monitor the results of the call (that or we don't care).  That being said, let me look at it again and see if it makes sense to still include the event listeners.

mttwc commented:
Just a question above.

leakymemory committed:
Clean things up a little more.  I tried to re-work things so that we could still detect errors
when the window auto-closed, but no matter what I did, Safari kept crashing. I played around with
it for a little while and found that just setting the event listener was enough to cause a crash. So,
in the cases where we don't need to monitor for the return URL, we won't have error information.

mttwc commented:
Would love to walk through this code on the non-Sierra machine with you today if possible. I want to understand this change a bit more.

mttwc commented on src/scripts/extensions/safari/safariWorker.ts:
```diff
@@ -129,34 +129,39 @@ export class SafariWorker extends ExtensionWorkerBase<SafariBrowserTab, SafariBr
+				});
+			}
+
+			newWindow.activeTab.url = url;
```
Nice. It makes way more sense for this line to be at the end when all the listeners have been attached. Not doing this in the past for things like setting img urls and whatnot has led to some hard-to-catch bugs, so we should continue to watch out for these in the future :)

leakymemory replied:
Agreed.
